### PR TITLE
Feat 202 modality change

### LINKED ISSUE #194: Raise exception if s3 bucket and path already exists
# User story
As a user, I want to have an automated check whether the s3 location exists before uploading, so I can avoid overwriting files unintentionally.

# Acceptance criteria
- [ ] When an s3 job is run, then there is a quick check to see if the s3 write location already exists.

# Sprint Ready Checklist 
- [ ] 1. Acceptance criteria defined
- [ ] 2. Team understands acceptance criteria
- [ ] 3. Team has defined solution / steps to satisfy acceptance criteria
- [ ] 4. Acceptance criteria is verifiable / testable
- [ ] 5. External / 3rd Party dependencies identified
- [ ] 6. Ticket is prioritized and sized

# Notes
*Add any helpful notes here.*

### LINKED ISSUE #202: add `experiment-type`
`aind-data-schema` changed:
- in directory/prefix names, `modality` has been replaced with `experiment_type`. 
- `modality` remains a field in `DataDescription`, however its controlled vocabulary has changed.

Update `aind-data-transfer` to handle this.

Acceptance Criteria:
- [ ] for the generic upload job, the command line arg `modality` should be renamed `experiment-type`
- [ ] for the generic upload job, the CSV schema should add a new column `experiment_type`. 
- [ ] for the generic upload job, the CSV schema should retain the modality column, but it should be populated with "abbreviations" from the `Modality` schema.

### PR DESCRIPTION
**Pull request by jtyoung84**
Closes #202
Closes #194 

- Adds experiment type and modality to base configs
- Uses pydantic for configuration settings
- Updates metadata creation for latest aind-data-schema version
- Creates an updated generic job that can be extended for ephys job later
- Adds ability for user define where temp directory is created
- Cleans up csv file parsing
- Allows for logging level to be set in individual jobs

### CODE REVIEW

File: src/aind_data_transfer/jobs/basic_job.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        )
+        subject_file_name = temp_dir / subject_metadata.output_filename
+        subject_metadata.write_to_json(subject_file_name)
+        procedures_metadata = ProceduresMetadata.from_service(
+            domain=self.configs.metadata_service_domain,
+            subject_id=self.configs.subject_id,
+        )
+        procedures_file_name = temp_dir / procedures_metadata.output_filename
+        procedures_metadata.write_to_json(procedures_file_name)
+        data_description_metadata = RawDataDescriptionMetadata.from_inputs(
```
Comment (dyf) [inline src/aind_data_transfer/jobs/basic_job.py:110]: do you need experiment type here?

File: src/aind_data_transfer/transformations/metadata_creation.py [hunk 2]
```diff
@@ -351,9 +355,11 @@ def from_inputs(
             if isinstance(funding_source, tuple)
             else funding_source
         )
-        data_description_instance = RawDataDescription.from_name(
-            name,
+        basic_settings = RawDataDescription.parse_name(name=name)
```
Comment (dyf) [inline src/aind_data_transfer/transformations/metadata_creation.py:358]: We can add `from_name` back if you want.

**Review by dyf — CHANGES_REQUESTED** [review]

File: src/aind_data_transfer/jobs/basic_job.py (hunk 1 above)
Comment (jtyoung84) [reply src/aind_data_transfer/jobs/basic_job.py:110]: It uses the parse_name method from RawDataDescription (line 300 in aind_data_schema.data_description) to pull exp type from the inputs.

File: src/aind_data_transfer/transformations/metadata_creation.py (hunk 2 above)
Comment (jtyoung84) [reply src/aind_data_transfer/transformations/metadata_creation.py:358]: I think it should be fine without any changes.

**Review by dyf — APPROVED** [review]

**Review by carshadi — APPROVED** [review]

**Pull request merged**